### PR TITLE
12.0 add my current timesheet

### PR DESCRIPTION
**Pull request by victor-champonnois**
Adding "My Current Timesheet Sheet" menu, triggering a server action that returns a view with the user's current timesheet.  
This view is also opened when clicking on the root menu.

I am not sure about the use of the `@api.model` decorator here. It was in [this repository](https://git.myceliandre.fr/Myceliandre/cooperative-market/src/branch/12.0/purchase_order_generator) that I used as a model, but I am not sure whether it should be used here.

Task : https://gestion.coopiteasy.be/web#id=7611&view_type=form&model=project.task&action=479

### CODE REVIEW

**Comment by huguesdk** [comment]
thanks for this feature, @coccinouille! my bad: i suggested `hr_timesheet_current` as module name, but a more correct name would be `hr_timesheet_sheet_current`, because it is related to the `hr_timesheet.sheet` model and not `hr_timesheet`.

File: hr_timesheet_current/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+{
+    'name': 'Hr Timesheet Current',
+    'description': """
+        Adds "my current timsheet" menu and open it directly when selecting the Timesheet menu""",
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Coop IT Easy SCRLfs',
+    'depends': [
+        'hr_timesheet',
```
Comment (huguesdk) [inline hr_timesheet_current/__manifest__.py]: this dependency is redundant, as `hr_timesheet_sheet` depends on it already.

File: hr_timesheet_current/views/hr_timesheet_sheet.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+    <record id="action_hr_timesheet_current_open" model="ir.actions.server">
+        <field name="name">My Current Timesheet Sheet</field>
+        <field name="model_id" ref="model_hr_timesheet_sheet" />
+        <field name="state">code</field>
+        <field name="code">
+            action = model.get_current_timesheet()
+        </field>
+    </record>
```
Comment (huguesdk) [inline hr_timesheet_current/views/hr_timesheet_sheet.xml]: this action should be defined before the menuitem.

File: hr_timesheet_current/views/hr_timesheet_sheet.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <menuitem
+        action="action_hr_timesheet_current_open"
+        id="menu_act_hr_timesheet_sheet_form_my_current"
+        name="My current Timesheet Sheet"
```
Comment (huguesdk) [inline hr_timesheet_current/views/hr_timesheet_sheet.xml]:
```suggestion
        name="My Current Timesheet Sheet"
```
it seems that title case is the norm (sadly 😉) for menu items in odoo.

File: hr_timesheet_current/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Victor Champonnois
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Hr Timesheet Current',
```
Comment (huguesdk) [inline hr_timesheet_current/__manifest__.py]:
```suggestion
    'name': 'My Current Timesheet Sheet',
```
or any other better-sounding name. ☺

File: hr_timesheet_current/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    'description': """
+        Adds "my current timsheet" menu and open it directly when selecting the Timesheet menu""",
```
Comment (huguesdk) [inline hr_timesheet_current/__manifest__.py]:
```suggestion
    'summary': """
        Allow to access the current timesheet sheet directly from the menu""",
```
`summary` is what is displayed in the apps list.

File: hr_timesheet_current/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    'depends': [
+        'hr_timesheet',
+        'hr_timesheet_sheet',
+    ],
+    'data': [
+        "views/hr_timesheet_sheet.xml"
+    ],
+    'demo': [
+    ],
+}
```
Comment (huguesdk) [inline hr_timesheet_current/__manifest__.py]: please pass this file through black.

**Review by huguesdk — CHANGES_REQUESTED** [review]
thanks a lot, it works like a charm!

please add a readme using the [oca maintainer tools readme generator](https://github.com/OCA/maintainer-tools#readme-generator).

File: hr_timesheet_current/views/hr_timesheet_sheet.xml (hunk 3 above)
Comment (huguesdk) [reply hr_timesheet_current/views/hr_timesheet_sheet.xml:7]: or, actually, just “Current Timesheet Sheet” would already be enough. but the choice is yours!

File: hr_timesheet_current/__manifest__.py (hunk 4 above)
Comment (huguesdk) [reply hr_timesheet_current/__manifest__.py]: or, actually, just “Current Timesheet Sheet” would already be enough. but the choice is yours!

**Comment by victor-champonnois** [comment]
Thanks @huguesdk, I added your requests !

**Review by huguesdk — APPROVED** [review]
thanks @coccinouille!

**Pull request merged**